projects/packages/videopress/src/client/state/reducers.js

```diff
@@ -79,8 +79,19 @@ const videos = ( state = {}, action ) => {
 		case SET_VIDEO: {
 			const { video } = action;
 			const { items = [] } = state;
-			if ( ! items.find( item => item.ID === video.ID ) ) {
+
+			if ( ! items.length ) {
+				// Add video when no items
 				items.push( video );
+			} else {
+				const videoIndex = items.findIndex( item => item.id === video.id );
+				if ( videoIndex === -1 ) {
+					// Add video when not found
+					items.push( video );
+				} else {
+					// Update video when found
+					items[ videoIndex ] = video;
+				}
```

Comment:
There's no need to check against `items.length`, since the default is an empty array `[]`, when you check `findIndex` against an empty array it will return `-1` and goes to the first `if`

Reply:
I check `items.length` to avoid calling the `findIndex` method. It happens in a hard refresh where no items are in the store. If so, it just push the item.

Reply:
After an excellent talk, we agreed that the best approach is to rely on findIndex and avoid checking the items.length.

https://jsben.ch/SdWJh